loglib/qlog.c

```diff
@@ -382,12 +379,19 @@ int qlog_packet_frame(bytestream * s, void * ptr)
         if ((ftype & 4) != 0) {
             byteread_vint(s, &offset);
         }
-        uint64_t length = bytestream_remain(s);
-        if ((ftype & 2) != 0) {
-            byteread_vint(s, &length);
-        }
+        uint64_t length = 0;
```

Comment:
I think I understand where this comes from. We may not write the whole packet into the binary logfile and hence the bytestream_remain does not reflect the original packet length. Nevertheless, the length property is only available when the LEN bit was set to 1.

Reply:
There is an ongoing discussion of this in the quicdev slack, in the qlog channel. Robin Marx has opened two related issues for the qlog spec:

* https://github.com/quiclog/internet-drafts/issues/54
* https://github.com/quiclog/internet-drafts/issues/56

The qlog processing apps expect to see a length field for the stream frame, either set from the length parameter if present, or computed from the number of remaining bytes in the packet if absent. I ended up with logging two fields: a length frame that will work with qvis and other apps, and a boolean indicating whether the length was specified.

Reply:
Thanks Christian! The explanation helps a lot.

Still, my complaint about attempting to read the length from the byte stream when it clearly wasn't written there is something we shouldn't do.

Reply:
When the length parameter is not set, the effective length is from the first byte after the stream header until the end of the packet. We know the end of the packet, so we can compute that.

Reply:
What I am tryting to point out is that
```
byteread_vint(s, &length);
```
should still be
```
if ((ftype & 2) == 0) {
    byteread_vint(s, &length);
}
```

It probably works right now because the stream doesn't contain any data so this doesn't change the content of `&length` if the LEN flag was 0.

Reply:
Did you look at the corresponding code in logwriter.c? The frame in the binlog is not a straight copy of the frame on the wire. It is guaranteed to have a length field.

Reply:
No I missed that. Thx for pointing that out!